subprojects/cli/src/main/java/org/gradle/cli/CommandLineParser.java

```diff
@@ -454,18 +454,14 @@ public ParserState onComplete() {
             if (getHasArgument() && values.isEmpty()) {
                 throw new CommandLineArgumentException(String.format("No argument was provided for command-line option '%s'.", optionString));
             }
-            
+
             ParsedCommandLineOption parsedOption = commandLine.addOption(optionString.option, option);
             if (values.size() + parsedOption.getValues().size() > 1 && !option.getAllowsMultipleArguments()) {
                 throw new CommandLineArgumentException(String.format("Multiple arguments were provided for command-line option '%s'.", optionString));
             }
             for (String value : values) {
                 parsedOption.addArgument(value);
             }
-            if (option.getDeprecationWarning() != null) {
```

Comment:
Why is this removed? If so, the deprecation message would no longer make sense. https://github.com/gradle/gradle/blob/8c091930436481fc73bc4bf6c9ab7b397b31b60b/subprojects/build-option/src/main/java/org/gradle/internal/buildoption/CommandLineOptionConfiguration.java#L94

Reply:
It does still make sense, because it is shown in the `gradle help` output.

I removed this log message in this place, because it was not the right place for several reasons:

- it could be logged before our logging infrastructure was initialized, meaning the message got lost in embedded mode
- it was only logged when used with the CLI switch, but not logged when used with the Gradle property

Reply:
Maybe we could remove the unused `deprecationPrinter` property and related constructor of `CommandLineParser` class?

Reply:
Yes, you're right I missed that.